platform/x11/detect.py

```diff
@@ -149,6 +149,18 @@ def configure(env):
     env.Append(CCFLAGS=['-pipe'])
     env.Append(LINKFLAGS=['-pipe'])
 
+    # Check for gcc version > 4 before adding -no-pie
+    import re
+    import subprocess
+    proc = subprocess.Popen([env['CXX'], '--version'], stdout=subprocess.PIPE)
```

Comment:
What happens if CXX is `clang++`? Should we also restrict `-fpie` to Clang 5+?

For the reference, I had a slightly simpler hack for this test in the 2.1 branch: https://github.com/godotengine/godot/blob/2.1/platform/x11/detect.py#L94-L103
Namely `-dumpversion` can spare you a regexp.

Reply:
Clang 4.0 also doesn't supports `-no-pie` so it seems right to leave the version check as it is now.
`-dumpversion` seems to return always 4.2.1 in all the clang versions I've checked (I checked 4.0, 5.0 and 6.0).

Reply:
Can't we do what auto tools does in this situation? Run the compiler with an empty c file with the option and see if it errors out?

Reply:
I'd go with whatever is simpler and the least amount of lines of code :)

Reply:
@hpvb My first idea was to use TryCompile with this but if it failed I couldn't find a way to actually remove the added flag. So I went with this compiler check that worked.